### DISCUSSION TITLE
OpenApi for your REST APIs in NestJS

### DISCUSSION THREAD

**Discussion by giscus** [notiz-dev/notiz#176] [General]
# OpenApi for your REST APIs in NestJS

Setup Swagger to generate an OpenApi documentation for your REST endpoints.

https://notiz.dev/blog/openapi-in-nestjs

**Comment by coding-to-music** [comment]
Hi, cool tutorial, I really like Swagger now, thanks!
I was not able to get the final part to work, the Before / After, I don't think the After is correct.
My repo is here: https://github.com/coding-to-music/swagger-nestjs

Also, there is mention of Product - those probably were typos and you meant Users
Thanks

**Reply by coding-to-music** [reply]
Not sure what happened, but now I get this error, not sure how to figure out even where it is happening
npm run start:dev

> swagger-nestjs@0.0.1 start:dev
> nest start --watch


 Error  Unexpected token e in JSON at position 1

**Reply by marcjulian** [reply]
Thanks for catching my typo, changed it from product to user. I think i started out with product ... 🙄 
Did you fix that error? Otherwise to do have a reproduction repo?

**Reply by coding-to-music** [reply]
I don't, looks like I did not continue with it, thanks though

**Comment by indrasenakatam** [comment]
Can you please also help how to add http @Options() http method for any of the method PATCH, DELETE,PUT method. And how we can expose that @Options() route in SWAGGER. 

I am also using same @nestjs/swagger and enabled **CORS**  to support OPTIONS method (preflight browser request).
All are working but in my case I need to **expose** Options method for each method in Swagger file. 

Thank you very much in advance.

**Comment by nson22** [comment]
It's possible download the schema created?
